### PR TITLE
AP_RangeFinder: fixed lightware serial with LW20 lidar

### PR DESCRIPTION
**Pull request by tridge**
it needs a longer serial write to force it to serial mode from i2c

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp [hunk 1]
```diff
@@ -33,7 +34,10 @@ AP_RangeFinder_LightWareSerial::AP_RangeFinder_LightWareSerial(RangeFinder::Rang
 {
     uart = serial_manager.find_serial(AP_SerialManager::SerialProtocol_Rangefinder, serial_instance);
     if (uart != nullptr) {
-        uart->begin(serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_Rangefinder, serial_instance));
+        baudrate = serial_manager.find_baudrate(AP_SerialManager::SerialProtocol_Rangefinder, serial_instance);
+        uart->begin(baudrate);
+        // send enough serial transitions to trigger LW20 into serial mode
+        uart->write("www\r\n");
```
Comment (rmackay9) [inline libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: should last_init_ms be set here as well?

**Comment by rmackay9** [comment]
really great to see this.  there have been a number of users complaining.

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: doesn't actually matter

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp [hunk 2]
```diff
@@ -45,6 +49,7 @@ AP_RangeFinder_LightWareSerial::AP_RangeFinder_LightWareSerial(RangeFinder::Rang
 bool AP_RangeFinder_LightWareSerial::detect(AP_SerialManager &serial_manager, uint8_t serial_instance)
 {
     return serial_manager.find_serial(AP_SerialManager::SerialProtocol_Rangefinder, serial_instance) != nullptr;
+
```
Comment (OXINARF) [inline libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: Left-over?

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp (hunk 1 above)
Comment (OXINARF) [reply libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: It doesn't, but we might be repeating the same actions every time it boots (it just depends on the `update` being called in less than one second or not).

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp [hunk 3]
```diff
@@ -74,8 +79,17 @@ bool AP_RangeFinder_LightWareSerial::get_reading(uint16_t &reading_cm)
...
 
-    // we need to write a byte to prompt another reading
-    uart->write('d');
+    uint32_t now = AP_HAL::millis();
+    if (now - last_init_ms > 1000 &&
+        now - state.last_reading_ms > 1000) {
+        // send enough serial transitions to trigger LW20 into serial
+        // mode
+        uart->write("www\r\n");
+        uart->begin(baudrate, 256, 256);
```
Comment (OXINARF) [inline libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: Is there a reason why we "force" the buffer size here but not in init?

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: fixed

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: fixed

File: libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp (hunk 3 above)
Comment (tridge) [reply libraries/AP_RangeFinder/AP_RangeFinder_LightWareSerial.cpp]: i've confirmed it isn't needed (was added while debugging)

**Pull request merged**